editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/component/wizard/ComponentManager.java

```diff
@@ -74,5 +75,16 @@ private void initTagMap(CamelModel model) {
 	public Set<Component> getAllComponents() {
 		return new HashSet<>(model.getComponents());
 	}
+	
+	public Component getComponentById(String id) {
+		Iterator<Component> iter = model.getComponents().iterator();
```

Comment:
optional: it i s possible to write it with lambda  with fewer lines of code
something like 
```
return model.getComponents().stream()
				.filter(component -> id.equalsIgnoreCase(component.getId())).findAny()
				.orElse(null);
```